ir/id.h

```diff
@@ -48,10 +48,13 @@ struct ID : Util::IHasSourceInfo {
     }
     bool operator==(const ID &a) const { return name == a.name; }
     bool operator!=(const ID &a) const { return name != a.name; }
+    bool operator<(const ID &a) const { return name < a.name; }
```

Comment:
hrm... `operator<` for `cstring` actually does a lexicographical order. Do we want this behavior here? On the other hand, order by pointer address is not stable across different compiler invocations.

Reply:
Using cstring's lexicographical order was the intention here. I would expect comparing two IDs to have the same result as comparing two cstrings. 

I can add a comment?

Reply:
Yeah, comment would be fine.